essential-projects/caching
# Feature/fix test typings

## What did you change?

- Added @types/mocha to the devDependencies
- Made TCachedObject optional by giving it a default value of `any`

without these changes, when running `npm test` we would get these warnings:

```
[06:04:13] Starting 'test-typescript-build'...
[gulp-typescript] 2304 Cannot find name 'describe'.
> file: /Users/heiko/Desktop/Projekte/BPMN/process_engine_meta/caching/test/cache_repository.ts:
> [5] describe('CacheRepository', () => {
[gulp-typescript] 2707 Generic type 'CacheRepository<TCachedObject, TCacheWrapper>' requires between 1 and 2 type arguments.
> file: /Users/heiko/Desktop/Projekte/BPMN/process_engine_meta/caching/test/cache_repository.ts:
> [6]   let cacheRepository: CacheRepository;
[gulp-typescript] 2304 Cannot find name 'beforeEach'.
> file: /Users/heiko/Desktop/Projekte/BPMN/process_engine_meta/caching/test/cache_repository.ts:
> [8]   beforeEach((): void => {
[gulp-typescript] 2304 Cannot find name 'afterEach'.
> file: /Users/heiko/Desktop/Projekte/BPMN/process_engine_meta/caching/test/cache_repository.ts:
> [12]   afterEach(() => {
[gulp-typescript] 2304 Cannot find name 'it'.
> file: /Users/heiko/Desktop/Projekte/BPMN/process_engine_meta/caching/test/cache_repository.ts:
> [16]   it('should garbage collect after expiration from set', (next: any): void => {
[gulp-typescript] 2304 Cannot find name 'it'.
> file: /Users/heiko/Desktop/Projekte/BPMN/process_engine_meta/caching/test/cache_repository.ts:
> [37]   it('should garbage collect after expiration from config', (next: any): void => {
[gulp-typescript] 2304 Cannot find name 'it'.
> file: /Users/heiko/Desktop/Projekte/BPMN/process_engine_meta/caching/test/cache_repository.ts:
> [60]   it('should garbage collect after expiration from default', (next: any): void => {
[gulp-typescript] 2304 Cannot find name 'it'.
> file: /Users/heiko/Desktop/Projekte/BPMN/process_engine_meta/caching/test/cache_repository.ts:
> [79]   it('should return cache shortly before garbage collecting', (next: any): void => {
[gulp-typescript] 2304 Cannot find name 'it'.
> file: /Users/heiko/Desktop/Projekte/BPMN/process_engine_meta/caching/test/cache_repository.ts:
> [100]   it('should not return cache after removal', (next: any): void => {
[gulp-typescript] 2304 Cannot find name 'it'.
> file: /Users/heiko/Desktop/Projekte/BPMN/process_engine_meta/caching/test/cache_repository.ts:
> [118]   it('should not garbage collect after validity has been extended through usage', (next: any): void => {
[gulp-typescript] 2304 Cannot find name 'it'.
> file: /Users/heiko/Desktop/Projekte/BPMN/process_engine_meta/caching/test/cache_repository.ts:
> [147]   it('should not garbage collect if garbage collection is disabled', (next: any): void => {
[gulp-typescript] 2304 Cannot find name 'it'.
> file: /Users/heiko/Desktop/Projekte/BPMN/process_engine_meta/caching/test/cache_repository.ts:
> [171]   it('should garbage collect during validation', (next: any): void => {
TypeScript: 12 semantic errors
TypeScript: emit succeeded (with errors)
```

(see the red text on the red background [here](http://ci.process-engine.io:8080/blue/organizations/jenkins/essential-projects_node-lts%2Fcaching/detail/master/50/pipeline#step-77-log-32))

## How can others test the changes?

- clone the repo
- run `npm install`
- run `npm test`
- See how you get warnings without this fix, and don't warnings with this fix

## PR-Checklist

Please check the boxes in this list after submitting your PR:
